test/distributed/org/apache/cassandra/distributed/test/trackwarnings/AbstractClientSizeWarning.java

```diff
@@ -283,15 +295,26 @@ public boolean matches(InetAddress value)
         }
         assertHistogramUpdated();
         assertWarnAborts(0, 2, 1);
+    }
+
+    public void failThresholdDisabled(String cql) throws UnknownHostException
+    {
+        ICoordinator node = CLUSTER.coordinator(1);
+        for (int i = 0; i < failThresholdRowCount(); i++)
+            node.execute("INSERT INTO " + KEYSPACE + ".tbl (pk, ck, v) VALUES (1, ?, ?)", ConsistencyLevel.ALL, i + 1, bytes(512));
+
+        if (shouldFlush())
+            CLUSTER.stream().forEach(i -> i.flush(KEYSPACE));
 
         // query should no longer fail
         enable(false);
+        checkpointHistogram();
         SimpleQueryResult result = node.executeWithResult(cql, ConsistencyLevel.ALL);
         assertThat(result.warnings()).isEmpty();
         assertHistogramNotUpdated();
```

Comment:
One minor query. In Line 313, we are asserting that the warnings is empty. Isn't that sufficient? Do we still need to check if the histogram hasn't updated? Isn't it redundant?

Reply:
I didn't want to change current behavior. As you can see in the original test:
https://github.com/apache/cassandra/blob/trunk/test/distributed/org/apache/cassandra/distributed/test/trackwarnings/AbstractClientSizeWarning.java#L206-L207

It was asserting both things already. Same goes for the failThreshold method:
https://github.com/apache/cassandra/blob/trunk/test/distributed/org/apache/cassandra/distributed/test/trackwarnings/AbstractClientSizeWarning.java#L290-L291